addons/Popochiu/Editor/Importers/Aseprite/AsepriteController.gd

```diff
@@ -0,0 +1,222 @@
+	var only_visible_layers = options.get('only_visible_layers', false)
+	var basename = _get_file_basename(file_name)
+	var layers = list_layers(file_name, only_visible_layers)
+	var exception_regex = _compile_regex(exception_pattern)
+
+	var output = []
+
+	for layer in layers:
+		if layer != "" and (not exception_regex or exception_regex.search(layer) == null):
+			output.push_back(export_layer(file_name, layer, output_folder, options))
+
+	return output
+
+
+func export_layer(file_name: String, layer_name: String, output_folder: String, options: Dictionary) -> Dictionary:
+	var output_prefix = options.get('output_filename', "").strip_edges()
+	var output_dir = output_folder.replace("res://", "./").strip_edges()
+	var data_file = "%s/%s%s.json" % [output_dir, output_prefix, layer_name]
+	var sprite_sheet = "%s/%s%s.png" % [output_dir, output_prefix, layer_name]
+	var output = []
+	var arguments = _export_command_common_arguments(file_name, data_file, sprite_sheet)
+	arguments.push_front(layer_name)
+	arguments.push_front("--layer")
+	
+	_add_sheet_type_arguments(arguments, options)
+
+	var exit_code = _execute(arguments, output)
+	if exit_code != 0:
+		print('aseprite: failed to export layer spritesheet')
+		print(output)
+		return {}
+
+	return {
+		'data_file': data_file.replace("./", "res://"),
+		"sprite_sheet": sprite_sheet.replace("./", "res://")
+	}
+
+
+func list_layers(file_name: String, only_visible = false) -> Array:
+	var output = []
+	var arguments = ["-b", "--list-layers", file_name]
+
+	if not only_visible:
+		arguments.push_front("--all-layers")
+
+	var exit_code = _execute(arguments, output)
+
+	if exit_code != 0:
+		printerr('aseprite: failed listing layers')
+		printerr(output)
+		return []
+
+	return _sanitize_list_output(output)
+
+
+func list_tags(file_name: String) -> Array:
+	var output = []
+	var arguments = ["-b", "--list-tags", file_name]
+
+	var exit_code = _execute(arguments, output)
+
+	if exit_code != 0:
+		printerr('aseprite: failed listing tags')
+		printerr(output)
+		return []
+
+	return _sanitize_list_output(output)
+
+
+func is_valid_spritesheet(content):
+	return content.has("frames") and content.has("meta") and content.meta.has('image')
+
+
+func get_content_frames(content):
+	return content.frames if typeof(content.frames) == TYPE_ARRAY  else content.frames.values()
+
+
+func check_command_path():
+	# On Linux, MacOS or other *nix platforms, nothing to do
+	if not OS.get_name() in ["Windows", "UWP"]:
+		return true
+	
+	# On Windows, OS.Execute() calls trigger an uncatchable
+	# internal error if the invoked executable is not found.
+	# Since the error is unclear, we have to check that the aseprite
+	# command is given as a full path and return an error if it's not.
+	var regex = RegEx.new()
+	regex.compile("^[A-Z|a-z]:[\\\\|\\/].+\\.exe$")
+	return \
+		regex.search(_aseprite_command()) \
+		and \
+		File.new().file_exists(_aseprite_command())
+
+
+func test_command():
```

Comment:
I'm not sure how to do it actually. I have to understand how to hook into settings editing to show a popup. I'll study this solution and let you know.

Reply:
Now that I think more about this. It could be not a popup but a message in the Project settings > Popochiu section. Something that appears below the field where devs put the path to Aseprite, like: `( ! ) Aseprite path set successfully`.

Reply:
I agree. My main concern at the moment is that there is nothing that call your attention to look at the console to know what is going one.

Another thing we could do is to display the Godot warning icon (`get_icon("NodeWarning", "EditorIcons")` or `get_icon("StatusWarning", "EditorIcons")`) somewhere in the Aseprite section to let developers know that something is needed. Then we can add a message in a `hint_tooltip`. Something similar to what happens when you don't put a polygon inside an Area2D.

Reply:
Or if we want something simpler, the message on the console could be an error message

Reply:
Sorry, I just noticed there is already an error message that prints before the one you use to mention what devs should do to set de Aseprite command.

Reply:
Maybe I can add an error to the console to explain the subsequent one (that's triggered by Godot itself)?

I also like the warning sign in the interface: maybe I can show it in place of the "Import" button when the command is not verified? It's a bit heavy-handed because you can NOT import until you don't have a full path but things being what they are...

Reply:
I like that idea